src/lighteval/models/endpoint_model.py

```diff
@@ -41,17 +46,26 @@
 from lighteval.data import GenerativeTaskDataset, LoglikelihoodDataset
 from lighteval.logging.hierarchical_logger import hlog, hlog_err, hlog_warn
 from lighteval.models.abstract_model import LightevalModel, ModelInfo
-from lighteval.models.model_config import InferenceEndpointModelConfig, InferenceModelConfig
-from lighteval.models.model_output import GenerativeResponse, LoglikelihoodResponse, LoglikelihoodSingleTokenResponse
+from lighteval.models.model_config import EnvConfig, InferenceEndpointModelConfig, InferenceModelConfig
+from lighteval.models.model_output import (
+    GenerativeResponse,
+    LoglikelihoodResponse,
+    LoglikelihoodSingleTokenResponse,
+)
 from lighteval.tasks.requests import (
     GreedyUntilRequest,
     LoglikelihoodRequest,
     LoglikelihoodRollingRequest,
     LoglikelihoodSingleTokenRequest,
+    Request,
 )
 from lighteval.utils.utils import EnvConfig, as_list
 
 
+EndpointInput: TypeAlias = TextGenerationInput | ChatCompletionInput
+EndpointOutput: TypeAlias = TextGenerationOutput | ChatCompletionOutput
```

Comment:
The changes I made to endpoint model was to pave the way for the day Lighteval might add evaluation of commercial models, or add the evaluation of other base tasks e.g. visual question answering, reusing most of the logic in the parent endpoint model.  Endpoint model methods are organized as follows:

- `greedy_until()`, `loglikelihood()`, `loglikelihood_rolling()`: public apis of the model that could be reused in inheriting endpoint models. These methods call `_process_batch()` or `_async_process_batch()` 
- `_process_batch()` and `_async_process_batch()`: for batch processing and could be reused in inheriting endpoint models. They call `_prepare_request()` and then `_process_request()`.
- `_prepare_request()`: bears the responsibility to convert the incoming request to `EndpointInput` which is one of the `huggingface_hub.InferenceType` predefined types. This also could be reused among different endpoint classes.
- `_process_request()`: given the `EndpointInput`, it creates the `EndpointOutput` using the client. This is somewhat endpoint specific.
- `_process_generate_response()` and `_process_logprob_response()`: endpoint specific methods taking care of creating `ModelResponse` using the `EndpointOutput`. Before, these were part of the `greedy_until()` and `loglikelihood()` methods.


Specifically, I wanted to propose this directory structure for endpoint models:

```
lighteval/
    models/
        endpoints/
            endpoint_model.py
            inference_endpoint_model.py
            tgi_model.py
            anthropic_model.py
            openai_model.py
```
in which `endpoint_model.py` holds most of the logic and the child models override some methods if necessary.